Packages/MIES/MIES_GuiUtilities.ipf

```diff
@@ -1083,21 +1083,33 @@ Function RestoreCursors(graph, cursorInfos)
 	string graph
 	WAVE/T cursorInfos
 
-	string traceList, cursorTrace
-	variable i, numEntries
+	string traceList, cursorTrace, info, replacementTrace
+	variable i, numEntries, numTraces
 
 	traceList = TraceNameList(graph, ";", 0 + 1)
+	numTraces = ItemsInList(traceList)
+
+	if(numTraces == 0)
+		return NaN
+	endif
 
 	numEntries = DimSize(cursorInfos, ROWS)
 	for(i = 0; i < numEntries; i += 1)
+		info = cursorInfos[i]
 
-		cursorTrace = StringByKey("TNAME", cursorInfos[i])
+		if(IsEmpty(info)) // cursor was not active
+			continue
+		endif
+
+		cursorTrace = StringByKey("TNAME", info)
 
 		if(FindListItem(cursorTrace, traceList) == -1)
-			continue
+			// trace is not present anymore, use the first one instead
```

Comment:
@t-b is there a case(s) where the first trace isn't also the lowest number sweep?

Reply:
@timjarsky For the databrowser/sweepbrowser we now probably have the first trace being from the first sweep. But as that is a generic routine it does have no idea what the traces are about.